pkg/collector/corechecks/system/gpu/nvidia/jetson/jetson.go

```diff
@@ -119,56 +120,27 @@ func (c *JetsonCheck) processTegraStatsOutput(tegraStatsOuptut string) error {
 
 // Run executes the check
 func (c *JetsonCheck) Run() error {
-	// Kill tegrastats if it runs for twice as long as the interval we specified, to avoid blocking
-	// the check forever
-	ctx, cancel := context.WithTimeout(context.Background(), 2*tegraStatsInterval*time.Millisecond)
-	defer cancel()
-	cmd := exec.CommandContext(ctx, c.tegraStatsPath, c.commandOpts...)
-
-	// Parse the standard output for the stats
-	stdout, err := cmd.StdoutPipe()
-	if err != nil {
-		return err
+	tegraStatsCmd := fmt.Sprintf("%s %s", c.tegraStatsPath, strings.Join(c.commandOpts, " "))
+	cmdStr := fmt.Sprintf("(%s) & pid=$!; (sleep %d && kill -9 $pid)", tegraStatsCmd, int((2*tegraStatsInterval).Seconds()))
+	var cmd *exec.Cmd
+	if c.useSudo {
+		// -n, non-interactive mode, no prompts are used
+		cmd = exec.Command("sudo", "-n", "sh", "-c", cmdStr)
+	} else {
+		cmd = exec.Command("sh", "-c", cmdStr)
```

Comment:
Why did we change from `context.WithTimeout` to a manual `kill`? Can't we use it with `sudo`?

Reply:
Makes sense! Then I think I would try to report at least the kind of error that it seems we can identify (see comment below)